src/java.base/share/classes/java/lang/Process.java

```diff
@@ -612,6 +621,53 @@ public boolean waitFor(Duration duration) throws InterruptedException {
      */
     public abstract int exitValue();
 
+    /**
+     * Close all writer and reader streams and terminate the process.
+     * The streams are closed immediately and the process is terminated without waiting.
+     * <p>
+     * Before calling {@code close} the caller should read the streams for any
+     * data or text and call {@linkplain #waitFor() waitFor} if the exit value is needed.
+     * The contents of streams that have not been read fully are lost;
+     * they are discarded or ignored.
+     * Streams should be {@code closed} when no longer needed.
+     * Closing an already closed stream usually has no effect but is specific to the stream.
+     * Any {@code IOException} that occurs when closing a stream is ignored.
+     * <p>
+     * The process may already have exited or be in the process of exiting;
+     * if it is {@linkplain #isAlive() alive}, it is {@linkplain #destroy destroyed}.
+     * Any {@code IOException} that occurs when destroying the process is ignored.
+     * <p>
+     * Example using try-with-resources writing text to a process, reading back the
+     * response, and closing the streams and process:
+     * {@snippet class=ProcessExamples region=example}
+     *
+     * @implSpec
+     * The {@code outputWriter} and {@code outputStream} to the process are closed.
+     * The {@code inputReader} and {@code inputStream} from the process are closed.
+     * The {@code errorReader} and {@code errorStream} from the process are closed.
+     * The process is destroyed.
+     * @since 26
+     */
+    public void close() {
+        // Close each stream
+        quietClose(outputWriter != null ? outputWriter : getOutputStream());
+        quietClose(inputReader != null ? inputReader  : getInputStream());
+        quietClose(errorReader != null ? errorReader : getErrorStream());
```

Comment:
Async close of process stream is a complicated here. Do you have a summary on how it behaves on both Unix and Windows? I'm thinking about the deferred close code in particular. (I don't have an issue with close throwing away the exceptions, I'm just not 100% sure that async close is reliable in the first place).

> Closing an already closed stream usually has no effect but is specific to the stream.

I wonder if we could specify Process::close to be idempotent. That would mean we couldn't attempt to re-close if a first close of stream failed or had some other side effect.

Reply:
That would remove any question about behavior without having to delve into the details of every stream close.
Down at the bottom of every process stream is a FileDescriptor holding either an fd (Linux) or a handle (Windows).
This is true for all the redirects whether to files, pipes, or inherited (stdin, stdout, stderr). The FileDescritor.close method is very careful to impotently close and reset the fd/handle.